src/pudl/output/ferc1.py

```diff
@@ -3044,25 +3125,127 @@ def out_ferc1__yearly_rate_base(
         # the assets/liabilites both use ending_balance for its main $$ column
         .rename(columns={"dollar_value": "ending_balance"})
     )
-    # then select only the leafy exploded records that are in rate base and concat
-    in_rate_base = pd.concat(
+    return cash_working_capital_df
+
+
+def disaggregate_null_or_total_tag(
+    rate_base_df: pd.DataFrame,
+    tag_col: str,
+) -> pd.DataFrame:
+    """Disaggregate records with an null or total value in the ``tag_col``.
+
+    We have records in the rate base table with total and/or null values for
+    key tag columns which we want to separate into component parts because the
+    null or total values does not convery a level of detail we want for the
+    rate base table. This is done in two steps:
+
+    * :func:`get_tag_col_ratio` : for each ``report_year`` and ``utility_id_ferc1``,
+      get a ratio of all of the ``ending_balance`` for all of the non-null and
+      non-total tags.
+    * use this ratio to disaggregate the ``ending_balance`` from records with total
+      and null tags across the component tags.
+
```

Comment:
I still like the idea of saying "tag value" instead of just "tag" but this is non blocking